tests/test-portals.c

```diff
@@ -151,9 +240,18 @@ global_setup (void)
   g_mkdtemp (outdir);
   g_debug ("outdir: %s\n", outdir);
 
+  /* FIXME: Instead of this we should really use G_TEST_OPTION_ISOLATE_DIRS
+   * but it would require more rework to ensure that some desktop files are
+   * still found.
+   */
+  g_assert_no_errno (g_chdir (outdir));
```

Comment:
Urg. Could you expand?

Reply:
So.... Ideally I was still hacking to find a solution to avoid this and take the `G_TEST_OPTION_ISOLATE_DIRS` route, but in practice.

By using `G_TEST_OPTION_ISOLATE_DIRS` each single test has a separate working temporary directory, and I ideally would like to create our test-app-info file there, so that there's no risk that a test interfere with another one that is running afterwards.

However, right now some tests can't work with such flag, and so I kept using the `outdir` that has been used so far as the "test working directory". This is far from being ideal though, as we the have the right machinery to keep tests isolated without relying on a single place.